xcpmd/src/db-events-module.c

```diff
@@ -0,0 +1,229 @@
+    { "event_var_written_edge", IS_STATELESS, ARG_STR, { .str = "" }, EVENT_VAR_WRITTEN_EDGE }
+};
+
+static struct cond_table_row condition_data[] = {
+    { "whenVarWritten"   , var_was_written        , "s"   , "string var_name"  , EVENT_VAR_WRITTEN_EDGE , NULL } ,
+    { "intEqualTo"       , var_int_equals         , "i i" , "int a, int b"     , EVENT_VAR_WRITTEN      , NULL } ,
+    { "intGreaterThan"   , var_int_greater_than   , "i i" , "int a, int b"     , EVENT_VAR_WRITTEN      , NULL } ,
+    { "intLessThan"      , var_int_less_than      , "i i" , "int a, int b"     , EVENT_VAR_WRITTEN      , NULL } ,
+    { "floatGreaterThan" , var_float_greater_than , "f f" , "float a, float b" , EVENT_VAR_WRITTEN      , NULL } ,
+    { "floatLessThan"    , var_float_less_than    , "f f" , "float a, float b" , EVENT_VAR_WRITTEN      , NULL } ,
+    { "isTrue"           , var_is_true            , "b"   , "bool n"           , EVENT_VAR_WRITTEN      , NULL } ,
+    { "isFalse"          , var_is_false           , "b"   , "bool n"           , EVENT_VAR_WRITTEN      , NULL }
+};
+
+static unsigned int num_events = sizeof(event_data) / sizeof(event_data[0]);
+static unsigned int num_conditions = sizeof(condition_data) / sizeof(condition_data[0]);
+
+
+//Public data
+struct ev_wrapper ** _db_event_table;
+
+
+//Initializes the module.
+//The constructor attribute causes this function to run at load (dlopen()) time.
+__attribute__((constructor)) static void init_module() {
+
+    unsigned i;
+
+    //Allocate space for event tables.
+    _db_event_table = (struct ev_wrapper **)malloc(num_events * sizeof(struct ev_wrapper *));
+    if (!(_db_event_table)) {
+        xcpmd_log(LOG_ERR, "Failed to allocate memory\n");
+        return;
+    }
+
+    //Add all events to the event list.
+    for (i=0; i < num_events; ++i) {
+        struct event_data_row entry = event_data[i];
+        _db_event_table[entry.index]  = add_event(entry.name, entry.is_stateless, entry.value_type, entry.reset_value);
+    }
+
+    //Add all condition_types to the condition_type list.
+    for (i=0; i < num_conditions; ++i) {
+        struct cond_table_row entry = condition_data[i];
+        add_condition_type(entry.name, entry.func, entry.prototype, entry.pretty_prototype, _db_event_table[entry.event_index], entry.on_instantiate);
+    }
+}
+
+
+//Cleans up after this module.
+//The destructor attribute causes this to run at unload (dlclose()) time.
+__attribute__((destructor)) static void uninit_module() {
+
+    //Free event tables.
+    free(_db_event_table);
+}
+
+
+//Condition checkers
+bool var_was_written(struct ev_wrapper * event, struct arg_node * args) {
+
+    char *var_watched, *var_written;
+    
+    var_watched = get_arg(args, 0)->arg.str;
+    var_written = event->value.str;
+
+    if (var_watched == NULL || var_written == NULL) {
+        return false;
+    }
+
+    if ((strlen(var_watched) == strlen(var_written)) && strcmp(var_watched, var_written) == 0) {
```

Comment:
Just `strcmp(var_watched, var_written) == 0` should be fine - it'll take care of mismatched lengths.